### PR TITLE
Small refactoring for flexibility of subclasses (token stream, tree view)

### PR DESCRIPTION
**Pull request by parrt**

### CODE REVIEW

File: runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public Token get(int i) {
 		if ( start<0 || stop<0 ) return null;
 		lazyInit();
 		List<Token> subset = new ArrayList<Token>();
-		if ( stop>=tokens.size() ) stop = tokens.size()-1;
+		if ( stop>=size() ) stop = size()-1;
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java]: :exclamation: This is part of a bounds check for directly indexing `tokens`. The `tokens.get(i)` expression that follows needs to be updated to `this.get(i)` so the `size()` and `get(int)` methods are paired.

File: runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java [hunk 2]
```diff
@@ -235,9 +235,9 @@ public Token LT(int k) {
 
 		int i = p + k - 1;
 		sync(i);
-        if ( i >= tokens.size() ) { // return EOF token
+        if ( i >= size() ) { // return EOF token
             // EOF must be last token
-            return tokens.get(tokens.size()-1);
+            return tokens.get(size()-1);
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java]: :exclamation: This needs to be the following:

```java
return get(size() - 1);
```

File: runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java [hunk 3]
```diff
@@ -290,11 +290,11 @@ public void setTokenSource(TokenSource tokenSource) {
      */
     public List<Token> getTokens(int start, int stop, Set<Integer> types) {
         lazyInit();
-		if ( start<0 || stop>=tokens.size() ||
-			 stop<0  || start>=tokens.size() )
+		if ( start<0 || stop>=size() ||
+			 stop<0  || start>=size() )
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java]: :exclamation: These are currently bounds checks for direct access to `tokens`. (See previous comments for details.)

File: runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java [hunk 4]
```diff
@@ -468,7 +468,7 @@ public String getText(Interval interval) {
 		int stop = interval.b;
         if ( start<0 || stop<0 ) return "";
         lazyInit();
-        if ( stop>=tokens.size() ) stop = tokens.size()-1;
+        if ( stop>=size() ) stop = size()-1;
```
Comment (sharwell) [inline runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java]: :exclamation: This is currently a bound checks for direct access to `tokens`. (See previous comments for details.)

File: runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java (hunk 1 above)
Comment (parrt) [inline runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java]: makes sense

**Pull request merged**